### ISSUE TITLE
Cannot create text indexes

### ISSUE THREAD

**Issue by CircuitCoder** [mongodb-labs/mongo-rust-driver-prototype#223]
When creating text indexes, I get an error saying: "Index model keys must map to i32."

https://github.com/mongodb-labs/mongo-rust-driver-prototype/blob/master/src/coll/options.rs#L368

**Comment by saghm** [comment]
Hi! Can you provide some code that reproduces this?

**Comment by CircuitCoder** [comment]
According to the Javascript driver, a text index can be created in the following way, as said in http://mongodb.github.io/node-mongodb-native/2.2/tutorials/create-indexes/#create-a-text-index:
```javascript
coll.createIndex({
  field: 'text'
}, function callback() { /* ... */ });
```

But the following rust code:
```rust
coll.create_index(doc!{
    "field" => "text",
});
```

returns the error "Index model keys must map to i32.", because it only accepts integers for model keys.

Also, hashed indexes are created in the similar way, so I assume they currently cannot be created with the rust driver as well.

**Comment by CircuitCoder** [comment]
The C driver's implementation of index name generation is here:
https://github.com/mongodb/mongo-c-driver/blob/master/src/mongoc/mongoc-collection.c#L1029-L1043

It just accepts UTF8 values and concat them with the model key, like the way it deals with integers values.

So I guess it's safe to add a few match arms to allow string values like "text", "hashed" and so on.

**Comment by CircuitCoder** [comment]
I've just created a PR for this one.

**Comment by saghm** [comment]
Thanks! Sorry for not having a chance to get to it on my own quickly enough

**Comment by saghm** [comment]
Fixed in #228
